quinn-proto/src/crypto/rustls.rs

```diff
@@ -59,9 +61,9 @@ impl crypto::Session for TlsSession {
     }
 
     fn peer_identity(&self) -> Option<Box<dyn Any>> {
-        self.inner
-            .peer_certificates()
-            .map(|v| -> Box<dyn Any> { Box::new(v.to_vec()) })
+        self.inner.peer_certificates().map(|v| -> Box<dyn Any> {
```

Comment:
rustls gives us `CertificateDer<'_>` elements here which trivially convert from DER represented as `&[u8]` or `Vec<u8>`. How do you feel about just letting that leak into the public API? (Particularly given that rustls-pki-types is also intended as a long-term stable API.)

Reply:
rustls (when used) is a public dep no matter what, so I'm not opposed to exposing stuff from it. We're currently exposing a `rustls::Certificate` here anyway, after all.

Let's be sure to update the comment in `quinn::Connection::peer_identity` to reflect the actual type, if that's changing. Maybe also fill in the `_` in the `collect` call below for clarity? It's hard to tell what the type is after this PR.

Reply:
Decided to leave this for now, since for example a Noise `crypto::Session` still wouldn't want a `CertificateDer<'_>` to represent its identitifes. Did document the type for the rustls implementation.